shield/internal/parser/parser_test.go

```diff
@@ -100,6 +100,19 @@ func TestBooleanOperations(t *testing.T) {
 	require.Equal(t, "&&", and.Operator)
 }
 
+func TestNegativeInteger(t *testing.T) {
+	input := `-4;`
+	l := lexer.New(input)
+	p := New(l)
+
+	expression := p.Parse()
+	require.NotNil(t, expression)
+
+	negative, ok := ast.UnwrapPrefixExpression(expression)
+	require.True(t, ok, "expression is not *ast.PrefixExpression. got=%T", expression)
+	require.Equal(t, "-", negative.Operator)
+}
+
 func TestParser(t *testing.T) {
```

Comment:
can we add some tests here to ensure that the priority of operators is the expected one?